### ISSUE TITLE
Early Stopping for Neural NARX

### ISSUE THREAD

**Issue by wilsonrljr** [wilsonrljr/sysidentpy#108]
Early stopping is a technique used in preventing overfitting and improving generalization performance of neural network models. By monitoring the model performance on a validation set during the training process, early stopping provides a mechanism to halt the training when the model starts to exhibit signs of overfitting.

Besides, early stopping can contribute to saving computational resources and time. Training neural networks can be computationally expensive, especially with large datasets and complex architectures. Early stopping enables us to avoid unnecessary training iterations, reducing the computational burden and allowing for faster experimentation and model development. This efficiency makes early stopping a practical and valuable feature for neural network training.

The SysIdentPy maintainer (wilsonrljr) is committed to helping in all steps of the implementation to make Early Stopping available for the users.

**Comment by jamesyan20** [comment]
i'm interested in contributing to the project

**Comment by wilsonrljr** [comment]
Hey @jamesyan20, thanks! Can you send me a message on discord so we can talk to make a plan to work on this? You can find me by joining the SysIdentPy channel: https://discord.gg/8eGE3PQ

**Comment by Murad1997** [comment]
Hi @wilsonrljr, This Murad. I am interested in contributing to your project. If this issue has not been resolved I would be happy to contribute to it.

**Comment by wilsonrljr** [comment]
Hi @Murad1997 , thanks for your interest in contributing! This issue is not solved yet.

Would you like to continue our conversation here, or would you prefer to move it to Discord?

**Comment by Murad1997** [comment]
Hi @wilsonrljr, thanks for the reply. Sorry for the delayed response, as I was away from my PC. Let's discuss on the Discord.
